generator/src/main/java/com/scottlogic/deg/generator/generation/RegexStringGenerator.java

```diff
@@ -439,20 +479,24 @@ public boolean hasNext() {
 
         @Override
         public String next() {
-            currentIndex++; // starts at 1
-            return stringGenerator.getMatchedString(currentIndex);
+            String str = "";
+            do {
+                currentIndex++; // starts at 1
```

Comment:
"Do-while"s read weird to me. Maybe it's just personal preference but I think I would prefer to see this in a while loop and include the empty check in your `containsValidUtf8Chars` method above

Reply:
The do...while has moved, but I use it to reduce duplication of code.